pkg/cloud-init/cloud-init.go

```diff
@@ -202,12 +206,42 @@ func GenerateLocalData(vmiName string, hostname string, namespace string, source
 	} else {
 		return fmt.Errorf("userDataBase64 or userData is required for no-cloud data source")
 	}
+
+	var networkData []byte
+	var resolvData []byte
+	if source.NetworkData != "" {
+		parsedNetworkData := bytes.Split([]byte(source.NetworkData), []byte(cloudInitDelimiter))
```

Comment:
resolv_conf data is a main cloud-config object not an object of network-config (network). The resolve configuration you can add to network-config configures the dns configuration possible on some OSs like RedHat based ones. These are appended to resolv.conf and do not completely overwrite it. 

I take the network and resolv data and munge it together to pass to the cloud-init functions. It is then spit back out, if userData is of type cloud-config, it could be of type include, AND there isn't already a manage_resolv_conf instance it will append the resolv data to userData.